### PR TITLE
feat: [CDS-78501]: added UI validations for add webhook modal

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

- Added UI formik validations for the create/edit webhook modal

#### Screenshots

https://github.com/harness/harness-core-ui/assets/106532291/4cb2ce13-ba0b-49f1-bb5f-2c068cb3f8d5

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx [hunk 1]
```diff
@@ -85,7 +87,40 @@ export default function NewWebhookModal(props: NewWebhookModalProps): JSX.Elemen
   }
 
   return (
-    <Formik<AddWebhookModalData> initialValues={initialData} formName="addWebhookModalData" onSubmit={handleSubmit}>
+    <Formik<AddWebhookModalData>
+      initialValues={initialData}
+      formName="addWebhookModalData"
+      onSubmit={handleSubmit}
+      validationSchema={Yup.object().shape({
+        name: NameSchema(getString, {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx:95]: NIT: Use `NameIdentifierSchema` and then extend the shape.
Ref: https://github.com/harness/harness-core-ui/blob/develop/src/modules/72-triggers/components/Triggers/ManifestTrigger/ManifestWizardPageUtils.ts#L174

```
NameIdentifierSchema(getString, {
    nameRequiredErrorMsg: getString('common.validation.fieldIsRequired', {
            name: getString('name')
          })
  }).shape({
identifier: IdentifierSchema(getString),
        connectorRef: Yup.string().required(
          getString('common.validation.fieldIsRequired', {
            name: getString('platform.connectors.title.gitConnector')
          })
        ),
        repo: Yup.string().required(
          getString('common.validation.fieldIsRequired', {
            name: getString('repository')
          })
        ),
        folderPaths: Yup.array()
          .min(1)
          .of(
            Yup.object({
              value: Yup.string().required(
                getString('common.validation.fieldIsRequired', {
                  name: getString('common.git.folderPath')
                })
              )
            })
          )})
```

File: src/modules/75-cd/pages/webhooks/utils.tsx [hunk 2]
```diff
@@ -23,17 +23,16 @@ export interface NewWebhookModalProps {
   entityScope?: Scope
   closeModal: () => void
 }
+export function processFolderPaths(folderPaths: string[]): { id: string; value: string }[] {
+  return [...folderPaths].map(path => ({ id: uuid('', nameSpace()), value: path }))
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/pages/webhooks/utils.tsx:27]: NIT: This copy is not needed as map in immutable. 
```suggestion
  return folderPaths.map(path => ({ id: uuid('', nameSpace()), value: path }))
```

File: src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/NewWebhookModal.tsx]: done

File: src/modules/75-cd/pages/webhooks/utils.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/pages/webhooks/utils.tsx]: done

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**